### PR TITLE
fix: Resolve webpack dependencies

### PR DESCRIPTION
**Pull request by sendilkumarn**
**What kind of change does this PR introduce?**

tries to merge master on top of #229  Test cases seems to work fine. There is an extra line in the log which is handled (needs review)

@ev1stensberg @sokra

### CODE REVIEW

**Comment by ev1stensberg** [comment]
Could you rebase this to Tobias's PR instead? Easier to debug

**Comment by sendilkumarn** [comment]
This is already on top of Tobias' change. 

`https://github.com/webpack/webpack-cli/pull/251/commits/b8eaedfb1a328ccb91d7ba49289a22c2614fe38a` 

is the different

File: bin/webpack.js [hunk 1]
```diff
@@ -432,7 +434,7 @@
 			}
 
 			if (argv.progress) {
-				var ProgressPlugin = require("webpack/lib/ProgressPlugin");
+				var ProgressPlugin = require("webpack").processPlugin;
```
Comment (sokra) [inline bin/webpack.js]: processPlugin -> ProgressPlugin

File: bin/ErrorHelpers.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
+/*
```
Comment (sokra) [inline bin/ErrorHelpers.js]: Seem to be weird to have `errorHelpers` and `ErrorHelpers`.

Comment (sendilkumarn) [reply bin/ErrorHelpers.js]: it should be `errorHelpers` right?

**Comment by sendilkumarn** [comment]
rename the file too?

**Comment by sendilkumarn** [comment]
well it is super exciting 👍 the file shows `errorHelpers` but in the commit it is `ErrorHelpers` is there a bug with github

Ha ha, use `git mv` 😝

File: bin/webpack.js [hunk 3]
```diff
@@ -441,10 +443,10 @@
 			}
 
 			if (outputOptions.infoVerbosity === "verbose") {
-				compiler.hooks.beforeCompile.tap("WebpackInfo", (compilation) => {
+				compiler.hooks.beforeCompile.tap("WebpackInfo", compilation => {
```
Comment (ev1stensberg) [inline bin/webpack.js]: I think maybe this is already commited in master

Comment (sendilkumarn) [reply bin/webpack.js]: addressed.

File: bin/errorHelpers.js [hunk 4]
```diff
@@ -4,54 +4,27 @@
...
+			stack.length = i;
 	return stack.join("\n");
 };
 
-exports.cutOffLoaderExecution = stack =>
-	exports.cutOffByFlag(stack, loaderFlag);
-
-exports.cutOffWebpackOptinos = stack =>
-	exports.cutOffByFlag(stack, webpackOptionsFlag);
+exports.cutOffWebpackOptions = (stack) => exports.cutOffByFlag(stack, webpackOptionsFlag);
```
Comment (ev1stensberg) [inline bin/errorHelpers.js]: Could you tell me what these things are doing?

**Comment by ev1stensberg** [comment]
Could you explain what you're doing and why @sendilkumarn ?

File: bin/errorHelpers.js (hunk 4 above)
Comment (sendilkumarn) [reply bin/errorHelpers.js]: @ev1stensberg the file name change is slightly confused things here. I guess, this was the old errorhelpers content, I think I have to take the latest from the bin/errorhelpers.js right ?

correct me if I am wrong

Comment (ev1stensberg) [reply bin/errorHelpers.js]: yep, that would be prefered, thought you did that, so that's why I'm asking

Comment (sendilkumarn) [reply bin/errorHelpers.js]: yup, the old errorhelper was from Tobias, but the current master errorHelper was having all the methods what was there and new methods were added on top of it. So I decided to have this version.

Comment (ev1stensberg) [reply bin/errorHelpers.js]: So if I understand: You've looked at Tobias's PR, that is where this implementation of ErrorHelper is from?

Comment (sendilkumarn) [reply bin/errorHelpers.js]: yeah... But when I merged with the master I saw yours  (current master) which has more methods. 

Tobias PR : 
https://github.com/webpack/webpack-cli/pull/251/commits/cf82edb1c35e2f25689d8e293562f2ee501ab761

Comment (ev1stensberg) [reply bin/errorHelpers.js]: Keep what Tobias refactored, see if there's any old calls using the old code in the CLI, I'll re-review after

**Comment by sendilkumarn** [comment]
is it waiting for @sokra 's review ?

**Comment by ev1stensberg** [comment]
Is everything set? I'll have another review later

**Comment by ev1stensberg** [comment]
Okay so had a look at Tobias's PR, looks good. Could you add `exec-in-dir` and its implementation from the PR?

**Comment by sendilkumarn** [comment]
@ev1stensberg It is there, the `exec-in-directory` will be there.

**Comment by ev1stensberg** [comment]
okay cap 👍

File: bin/convert-argv.js [hunk 5]
```diff
@@ -394,7 +394,7 @@ module.exports = function(...args) {
 				defineObject = {};
 			},
 			function() {
-				var DefinePlugin = require("webpack/lib/DefinePlugin");
+				var DefinePlugin = require("webpack").DefinePlugin;
```
[the remark this answers is not in the document]
Comment (ev1stensberg) [reply bin/convert-argv.js]: Rebase from master first

[the remark this answers is not in the document]
Comment (sendilkumarn) [reply bin/convert-argv.js]: good point both of you 👍 rebased and changed

**Comment by sendilkumarn** [comment]
@fokusferit I am not too sure on the test case, but added one that will test the corresponding method. Let me know if you think otherwise.

File: package.json [hunk 6]
```diff
@@ -22,7 +22,7 @@
     "precommit": "lint-staged",
     "prepare": "yarn format",
     "pretest": "yarn lint",
-    "test": "nyc jest",
+    "test": "nyc jest test/BinTestCases.test.js",
```
Comment (fokusferit) [inline package.json]: was there a need for this change?

File: test/binCases/errors/clean-webpack-options/stdin.js [hunk 7]
```diff
@@ -0,0 +1,13 @@
+"use strict";
```
Comment (fokusferit) [inline test/binCases/errors/clean-webpack-options/stdin.js]: 👍 I understand that testing it isolated doesn't make much sense, but having some test checking for correct error output is already better then nothing :)

**Comment by ematipico** [comment]
@ev1stensberg @fokusferit @sendilkumarn  Should be upgrade to `webpack-beta.1`? Apart this, I'd say the PR is good to me!

**Comment by sendilkumarn** [comment]
can we do better in testing the messages 😞 It is quite troublesome to update them every time.

**Comment by ev1stensberg** [comment]
@sendilkumarn agree. Think @EugeneHlushko did some nice and neat stuff there at an earlier PR, might be a good FC

**Pull request merged**